modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java

```diff
@@ -358,21 +335,27 @@ public Object doSaveExtended() throws NamingException, NotSupportedException, Sy
             tx.begin();
             startedTx = true;
         }
+
+        FormData formData = getContext().getForm();
+        String source = formData.getString("source");
         try {
-            getSnapshotManager().persistRuntimeSnapshot(getContext().getCoreSession(), distribLabel, otherProperties,
-                    filter);
+            getSnapshotManager().persistRuntimeSnapshot(getContext().getCoreSession(), formData.getString("name"),
+                    readFormData(formData), filter);
         } catch (NuxeoException e) {
             log.error("Error during storage", e);
             if (tx != null) {
                 tx.rollback();
             }
-            return getView("savedKO").arg("message", e.getMessage());
+            return getView("savedKO").arg("message", e.getMessage()).arg("source", source);
         }
         log.info("Snapshot saved.");
+
         if (tx != null && startedTx) {
             tx.commit();
         }
-        return getView("saved");
+        String redirectUrl = getContext().getBaseURL() + getPath();
+        log.debug("Path => " + redirectUrl);
```

Comment:
You can use log4j2 format feature to save some resources.
```suggestion
        log.debug("Path => {}",  redirectUrl);
```

Otherwise, we tend to use a colon follow by a space and directly is value, for example `The document: {docId} has been removed`. I don't know if we want to follow such pattern here?

Reply:
Indeed this is old logging, I only moved that line when factoring out duplicated code, and did not pay much attention to it, but i do not see the point of computing the redirectUrl just for logging since we're not redirecting to it anyway.
So i will just remove it :+1: